### DISCUSSION TITLE
UnsupportedApiVersion error while running Elsa workflow 2.14.1 with different projects

### DISCUSSION THREAD

**Discussion by Rathna314** [elsa-workflows/elsa-core#5585] [Q&A]
Hello all,

I am getting the following error

"error":{"code":"UnsupportedApiVersion","message":"The HTTP resource that matches the request URI 'http://localhost:#####' does not support the API version '1.0'.","innerError":null}}

We have upgraded all our projects to .Net 8.0 and I am running Elsa workflow API 2.14.1 version on Windows. The ELSA workflow API works fine when run as an individual project but when I it as a service with other projects it gives me the UnsupportedApiVersion error. Please help of what I need to update. I already tried some solutions for Startup.cs which were answered online.

**Comment by glime-ai** [comment]
To address the issue you're encountering with the `UnsupportedApiVersion` error, it's important to ensure that your Elsa Workflow API is correctly configured to handle the API versioning that your application is using. Since you mentioned that the error occurs when running Elsa as a service with other projects, it's possible that there's a conflict with the API versioning settings between the projects.

Here are a few steps you can take to troubleshoot and resolve the issue:

1. **Check API Versioning Configuration**: Ensure that the API versioning is configured correctly in your `Program.cs` file (since you're using .NET 8.0, there might not be a `Startup.cs` file). Look for any middleware configuration related to API versioning and make sure it's consistent across all projects.

2. **Review Route Configuration**: Verify that the routes for your API endpoints are set up correctly and that they include the expected API version. If you're using attribute routing, check the `[ApiVersion]` or `[Route]` attributes on your controller actions.

3. **Inspect Service Registration**: Make sure that the services required for API versioning are registered in the DI container. This includes services like `IApiVersionDescriptionProvider`.

4. **Compatibility with .NET 8.0**: Since you've upgraded to .NET 8.0, ensure that all the NuGet packages you're using, including Elsa Workflow API, are compatible with .NET 8.0. You may need to update some packages to their latest versions.

5. **Check for Conflicting Middleware**: If you have custom middleware or other API versioning libraries in your project, they might conflict with Elsa's API versioning. Review the middleware order and configurations to resolve any conflicts.

6. **Consult Elsa Documentation and GitHub Issues**: Review the official Elsa documentation and search the Elsa GitHub issues for similar problems and solutions. Other users may have encountered and resolved similar issues.

7. **Isolate the Problem**: Try to isolate the problem by creating a minimal reproduction of the issue. This can help you identify whether the problem is with Elsa's configuration or with the interaction between different services in your application.

If you continue to experience issues, consider providing more details about your configuration and the exact steps you've taken so far. This will help in diagnosing the problem more effectively.

Since you're using .NET 8.0, remember that the `Startup.cs` file is replaced by the new `Program.cs` approach, so any solutions referring to `Startup.cs` need to be adapted accordingly.

If you need further assistance, please provide additional details about your `Program.cs` configuration and any specific changes you've made related to API versioning.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
